File: solutions/cpp/0011.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int ret = 0;
        int l = 0;
        int r = height.size() - 1;

        while (l < r) {
            int h = min(height[l], height[r]);
            ret = max(ret, (r - l) * h);
            while (height[l] <= h && l < r) l++;
            while (height[r] <= h && l < r) r--;
        }

        return ret;
    }
};

File: solutions/python3/0011.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ret = 0
        l = 0
        r = len(height) - 1

        while l < r:
            h = min(height[l], height[r])
            ret = max(ret, (r - l) * h)
            while height[l] <= h and l < r:
                l += 1
            while height[r] <= h and l < r:
                r -= 1

        return ret